### DISCUSSION TITLE
About graphql-java-extended-validation directive naming convention

### DISCUSSION THREAD

**Discussion by wickedev** [graphql-java/graphql-java#2685] [General]
@bbakerman

First of all, thank you for the awesome project.

graphql-java-extended-validation uses camel case for directives starting with uppercase letters, but most directives in the GraphQL ecosystem seem to start with lowercase letters. Of course, java backend developers would convenient to follow the naming convention of Java annotations, but since schemas are shared with frontend developers as well as backends, how about starting with a lowercase letter at least for the directives on schema?

https://www.apollographql.com/docs/apollo-server/schema/directives/
https://relay.dev/docs/api-reference/graphql-and-directives

**Comment by bbakerman** [comment]
The naming was in fact inspired from JSR 380 Java Validation annotations.


https://www.baeldung.com/javax-validation

```
public class User {

    @NotNull(message = "Name cannot be null")
    private String name;

    @AssertTrue
    private boolean working;

    @Size(min = 10, max = 200, message 
      = "About Me must be between 10 and 200 characters")
    private String aboutMe;

    @Min(value = 18, message = "Age should not be less than 18")
    @Max(value = 150, message = "Age should not be greater than 150")
    private int age;

    @Email(message = "Email should be valid")
    private String email;

    // standard setters and getters 
}
```

The code itself is based on Java validation classes as well.  So yeah one could argue that it's not how others do it - but to Java developers one could argue it is how others do it.  Either way - the names are picked now - they wont be changed

**Reply by bbakerman** [reply]
> There are only two hard things in Computer Science: cache invalidation, naming things and off by one errors.


https://martinfowler.com/bliki/TwoHardThings.html

**Reply by wickedev** [reply]
It's not about changing the Java annotation naming convention, it's about the directive naming shown in the schema. I don't know if the my intent was well communicated, but if it is, you don't mind this comment

**Reply by bbakerman** [reply]
Yeah I guess I was saying I took the Java annotation naming from JSR 380 and used it for the graphql directives naming.

So in java you  there is a `@Size` annotation and in graphql there is a `@Size` directive that does a similar thing.

So that's where the naming came from.

And now that we have set the naming it would be a breaking change to change it - and hence we wont.
